extensions/Worker.Extensions.Shared/Configuration/ConfigurationExtensions.cs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions
+{
+    internal static class ConfigurationExtensions
```

Comment:
To address your concern we discussed offline: `PackageReference` is **not** transitive when using `SharedReference`. When `Worker.Extensions.Shared.csproj` includes a `PackageReference`, any projects point to this via a `SharedReference` will **not** implicitly get that `PackageReference` as well. The referencing projects must include the appropriate dependencies themselves. This means keeping the packages consistent between the shared project and the referencing projects is a manual effort by us devs. But mistakes will be readily apparent via build errors.

Including the `PackageReference` in `Worker.Extensions.Shared.csproj` is optional and is only for design time assembly resolution in VS / VS Code so that intellisense works properly. However, I don't think that is even functioning due to the use of `Microsoft.Build.NoTargets` SDK here. Definitely not a perfect experience, so it is fine to leave it out for now. I might find some time and see if I can give a more optimal development experience for these shared projects.

For example, I added `Microsoft.Extensioms.Configuration/7.0.0` to the shared project, then built and packed the Blob project (from your PR). Here are the nuget dependencies of that package:

![image](https://user-images.githubusercontent.com/5439812/217654389-d11f4fb6-623d-4074-aa9e-01b3e2f0cc7e.png)

Notice there is no `Microsoft.Extensions.Configuration (>= 7.0.0)`? That is because `Worker.Extensions.Storage.Blob` does **not** depend on it - even though `Worker.Extensions.Shared` does depend on it.